pyfaidx/__init__.py

```diff
@@ -514,8 +488,8 @@ def read_fai(self):
 
     def build_index(self):
         try:
-            with self._fasta_opener(self.filename, 'rb') as fastafile:
-                with open(self.indexname, 'w') as indexfile:
+            with self._fasta_opener(self.filename, 'r', compression=self._compression) as fastafile:
```

Comment:
@hardingnj The mode here should be 'rb' if possible so that later code can see the line terminator characters.